gems/holiday_jp/0.8/_test/test.rb

```diff
@@ -0,0 +1,6 @@
+require "holiday_jp"
+
+holidays = HolidayJp.between(Date.new(2010, 9, 14), Date.new(2010, 9, 21)) # => [#<HolidayJp::Holiday>]
+holidays.first.name # => 敬老の日
+holidays.first.date # => Mon, 20 Sep 2010
```

Comment:
They just test that `untyped` accepts all methods. Because `holidays` is `::Array[Holiday] | ::Array[untyped]`, then `holidays.first` is `Holiday? | untyped?`, which is the same as `untyped`.
Could you make them more meaningful by updating the type definition or the tests?

Reply:
@pocke 
Thank you for your comment.
I am new to rbs and am having a little trouble.

The `Holidayjp.between` method returns either a class of `HolidayJp::Holiday` or an **empty array**.
So if I run the `steep check` as `holidays.first.name` in my test, I get the following error
```
test.rb:4:15: [error] Type `(::HolidayJp::Holiday | nil)` does not have method `name`
│ Diagnostic ID: Ruby::NoMethod
│
└ holidays.first.name # => 敬老の日
```
As a solution, if I use the test safe navigation operator, the `steep check` succeeds, is this the right way?
```
holidays = HolidayJp.between(Date.new(2010, 9, 14), holidays = HolidayJp.between(Date.new(2010, 9, 14), Date.new(2010, 9, 21)) # => [#<HolidayJp::Holiday>]
holidays.first&.name
holidays.first&.date
```

Reply:
> As a solution, if I use the test safe navigation operator, the steep check succeeds, is this the right way?


Yes, it is one of the options to write the test.
You can also use `or raise` idiom, or `Array#[]`.

```ruby
holidays = ...
holiday = holidays.first or raise
holiday.name
```

```ruby
holidays = ...
holiday = holidays[0] # Array#[] is not nil-able.
holiday.name
```


Which way is better depends on the situation.

Reply:
Thank you for your comment.
I will try to write a test using `Array#[]`.
Could you please review it?